scripts/version/README.md

```diff
@@ -0,0 +1,54 @@
+# How sonar-dotnet version is set in our pipeline
+
+## Introduction
+
+During the pipeline run, we promote the following artifacts:
+
+1. `sonar-csharp-plugin-x.y.z.buildNumber.jar`
+1. `sonar-vbnet-plugin-x.y.z.buildNumber.jar`
+1. `SonarAnalyzer.CSharp.x.y.z.buildNumber.nupkg`
+1. `SonarAnalyzer.VisualBasic.x.y.z.buildNumber.nupkg`
+1. `SonarAnalyzer.CSharp.Styling.x.y.z.buildNumber.nupkg`
+1. `SonarAnalyzer.CFG.x.y.z.buildNumber.nupkg` (promoted only on repox)
+
+There are two parts to the versioning of these artifacts (see also [Semantic Versioning 2.0.0](https://semver.org/)):
+ - The semantic version `x.y.z`, that is set through the [`set-version.ps1`](./set-version.ps1) - see also [example PR](https://github.com/SonarSource/sonar-dotnet/pull/9194) and persisted in the repository.
+ - The `buildNumber` which is updated in the [Azure pipeline](../../azure-pipelines.yml).
+
+To set the version locally we run the [`set-version.ps1`](./set-version.ps1) script, which uses the `mvn set:version` command to update the java plugin's semantic version and the `ChangeVersion.proj` to update the analyzers' semantic version. Later, the `ChangeVersion.proj` and the `mvn set:version` (via the [`update-maven-version-steps.yml`](https://dev.azure.com/sonarsource/DotNetTeam%20Project/_git/pipelines-yaml-templates?path=/update-maven-version-steps.yml&version=GBmaster&line=20&lineEnd=21&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)) are called in the pipeline to update the build version of the artifacts.
+
+## 1. How version is set in *.jar artifacts
```

Comment:
I don't like the numbers. They are hard to maintain to keep them consistent and we don't really need them.
```suggestion
## How version is set in *.jar artifacts
```